src/pem/_core.py

```diff
@@ -208,21 +227,52 @@ class OpenSSHPrivateKey(PrivateKey):
     re.DOTALL,
 )
 
+# See https://tools.ietf.org/html/rfc4716
+_RFC4716_RE = re.compile(
```

Comment:
Are you sure you can't put it in the `_PEM_TO_CLASS` above? Seems like it's just `"SSH2 PUBLIC KEY": SSHPublicKey` and `"SSH2 ENCRYPTED PRIVATE KEY": SSHCOMPrivateKey`?

That would also remove the diff inside of parse entirely.

Reply:
The differences is between `-----BEGIN` and `---- BEGIN`. One has 5 hashes the other has 4 dashes + 1 space.

I didn't wanted to make the regex to hard to read.

But I guess that I can add a generic handling for RFC 4716 like PEM SSH crypto ... this is a mess.

And we have PuTTY own key format that has on PEM representation.

Reply:
In that case make the space part of the class patterns. It’s still less complexity than handling them separately.

Reply:
The problem is that I don't know how to express in a clean regex way that a `---- BEGIN` (with space)  should also end with `---- END` with space.

If i put the fifth value as optional space or dash , the library will also load files with invalid formats.

See my last commit... but I would prefer to merge 08b34f0